pkg/policyGovernance/security/imageScanning/ImageScanService.go

```diff
@@ -249,13 +257,40 @@ func (impl ImageScanServiceImpl) fetchImageExecutionHistoryMapByIds(historyIds [
 	}
 	return mapOfExecutionHistoryIdVsExecutionTime, nil
 }
+func (impl ImageScanServiceImpl) fetchResourceScanningDataForEnv(appId, envId int) (int, error) {
+	cdWfRunner, err := impl.cdWorkflowRepo.FindLatestCdWorkflowRunnerByEnvironmentIdAndRunnerType(appId, envId, bean4.CD_WORKFLOW_TYPE_DEPLOY)
+	if err != nil {
+		impl.Logger.Errorw("error in fetching latest cd workflow runner of type DEPLOY for appId and envId", "appId", appId, "envId", envId, "err", err)
+		return 0, err
+	}
+	var ciArtifactId int
+	if cdWfRunner.CdWorkflow != nil && cdWfRunner.CdWorkflow.CiArtifact != nil {
+		// for images built by us or provided in external ci case.
+		ciArtifactId = cdWfRunner.CdWorkflow.CiArtifactId
+		if cdWfRunner.CdWorkflow.CiArtifact.ParentCiArtifact > 0 {
+			//linked ci case, parent ci artifact already scanned
+			ciArtifactId = cdWfRunner.CdWorkflow.CiArtifact.ParentCiArtifact
+		}
+	}
+
+	return ciArtifactId, nil
+}
 
 func (impl ImageScanServiceImpl) FetchExecutionDetailResult(request *bean3.ImageScanRequest) (*bean3.ImageScanExecutionDetail, error) {
 	//var scanExecution *security.ImageScanExecutionHistory
 	var scanExecutionIds []int
 	var executionTime time.Time
 	imageScanResponse := &bean3.ImageScanExecutionDetail{}
 	isRegularApp := false
+	if request.AppId > 0 && request.EnvId > 0 {
+		artifactId, err := impl.fetchResourceScanningDataForEnv(request.AppId, request.EnvId)
+		if err != nil {
+			impl.Logger.Errorw("error while fetching scan execution result", "appId", request.AppId, "envId", request.EnvId, "err", err)
+			return nil, err
+		}
+		// setting latest artifact deploy on app and env
+		request.ArtifactId = artifactId
+	}
 	if request.ImageScanDeployInfoId > 0 {
```

Comment:
add todo, that if request.ImageScanDeployInfoId > 0 check is to be removed after few sprints

Reply:
DONE